src/main/java/com/idukbaduk/itseats/menu/service/MenuService.java

```diff
@@ -0,0 +1,67 @@
+package com.idukbaduk.itseats.menu.service;
+
+import com.idukbaduk.itseats.menu.dto.MenuInfoDto;
+import com.idukbaduk.itseats.menu.dto.MenuListRequest;
+import com.idukbaduk.itseats.menu.dto.MenuListResponse;
+import com.idukbaduk.itseats.menu.entity.Menu;
+import com.idukbaduk.itseats.menu.entity.enums.MenuStatus;
+import com.idukbaduk.itseats.menu.error.MenuErrorCode;
+import com.idukbaduk.itseats.menu.error.MenuException;
+import com.idukbaduk.itseats.menu.repository.MenuRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class MenuService {
+
+    private final MenuRepository menuRepository;
+
+    public MenuListResponse getMenuList(MenuListRequest request) {
+        if (request.getStoreId() == null) {
+            throw new MenuException(MenuErrorCode.STORE_ID_REQUIRED);
+        }
+
+        List<Menu> menus = menuRepository.findMenusByStore(
+                request.getStoreId(),
+                request.getMenuGroup(),
+                request.getKeyword()
+        );
+
+        if (menus.isEmpty()) {
+            throw new MenuException(MenuErrorCode.MENU_NOT_FOUND);
+        }
+
+        int totalMenuCount = menus.size();
+        int orderableMenuCount = (int) menus.stream()
+                .filter(m -> m.getMenuStatus() == MenuStatus.ON_SALE)
+                .count();
+        int outOfStockTodayCount = (int) menus.stream()
+                .filter(m -> m.getMenuStatus() == MenuStatus.OUT_OF_STOCK)
+                .count();
+        int hiddenMenuCount = (int) menus.stream()
+                .filter(m -> m.getMenuStatus() == MenuStatus.HIDDEN)
+                .count();
```

Comment:
여기서 공통되는 로직을 메서드로 분리해보는건 어떤가요?

```java
    private int getMenuCount(List<Menu> menus, MenuStatus menuStatus) {
        return (int) menus.stream()
                .filter(m -> m.getMenuStatus() == menuStatus)
                .count();
    }
```

메서드로 분리한다면 코드가 더욱 간결해지고 좋을 거 같아요 😀

```suggestion
        int totalMenuCount = menus.size();
        int orderableMenuCount = getMenuCount(menus, MenuStatus.ON_SALE);
        int outOfStockTodayCount = getMenuCount(menus, MenuStatus.OUT_OF_STOCK);
        int hiddenMenuCount = getMenuCount(menus, MenuStatus.HIDDEN);
```